packages/issm/package.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Issm(AutotoolsPackage):
+    """Ice-sheet and Sea-Level System Model"""
+
+    homepage = "https://issm.jpl.nasa.gov/"
+    svn = "https://issm.ess.uci.edu/svn/issm/issm/trunk"
+
+    version("develop")
+
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool", type="build")
+    depends_on("m4", type="build")
+
+    depends_on("mpi")
+    depends_on("triangle")
+    depends_on("petsc+metis+mumps+scalapack")
+    depends_on("m1qn3")
+
+    def autoreconf(self, spec, prefix):
+        autoreconf("--install", "--verbose", "--force")
+
+    def configure_args(self):
+        args = [
+          "--with-wrappers=no",
+          "--enable-debugging",
+          "--enable-development",
+          "--enable-shared",
+          "--without-kriging",
+        ]
+        args.append("--with-triangle-dir={0}".format(self.spec["triangle"].prefix))
+        args.append("--with-petsc-dir={0}".format(self.spec["petsc"].prefix))
+        args.append("--with-metis-dir={0}".format(self.spec["metis"].prefix))
+        args.append("--with-mumps-dir={0}".format(self.spec["mumps"].prefix))
+        args.append("--with-m1qn3-dir={0}".format(self.spec["m1qn3"].prefix.lib))
+        args.append("--with-mpi-include={0}".format(self.spec["mpi"].prefix.include))
+
+        args.append("CC=" + self.spec["mpi"].mpicc)
+        args.append("CXX=" + self.spec["mpi"].mpicxx)
+        args.append("FC=" + self.spec["mpi"].mpifc)
+        args.append("F77=" + self.spec["mpi"].mpif77)
```

Comment:
Chipping away at things, I think these compiler overrides are necessary unless the nci-openmpi package exports its paths differently. Otherwise, we get errors like the following:
```
libtool: link: /g/data/x77/ahg157/spack/lib/spack/env/intel/icpc -g -O2 -g -O2 -fPIC -std=c++11 -D_DO_NOT_LOAD_GLOBALS_ -g -O2 -o .libs/issm_slc.exe main/issm_slc-issm_slc.o  ./.libs/libISSMCore.so -L/g/data/x77/ahg157/spack/opt/spack/linux-rocky8-cascadelake/intel-2021.10.0/petsc-3.19.4-tvawatn/lib -L/g/data/x77/ahg157/spack/opt/spack/linux-rocky8-cascadelake/intel-2021.10.0/mumps-5.5.1-fnmjlzj/lib -L/g/data/x77/ahg157/spack/opt/spack/linux-rocky8-cascadelake/intel-2021.10.0/metis-5.1.0-4gtjfh2/lib -L/apps/openmpi/4.1.4/lib ./.libs/libISSMOverload.so /g/data/x77/ahg157/spack/opt/spack/linux-rocky8-cascadelake/intel-2021.10.0/m1qn3-3.3-vjufbp6/lib/libm1qn3.a /g/data/x77/ahg157/spack/opt/spack/linux-rocky8-cascadelake/intel-2021.10.0/m1qn3-3.3-vjufbp6/lib/libddot.a -lpetsc -ldl -ldmumps -lmumps_common -lpord -lparmetis -lmetis -Wl,-rpath,/g/data/x77/ahg157/spack/opt/spack/linux-rocky8-cascadelake/intel-2021.10.0/issm-develop-ddhchuj/lib
/bin/ld: main/issm-issm.o: undefined reference to symbol 'MPI_Comm_set_attr'
//apps/openmpi/4.1.4/lib/libmpi.so.40: error adding symbols: DSO missing from command line
make[3]: *** [Makefile:5068: issm.exe] Error
```

It's trying to do the MPI linking itself, but using `icpc`, and completely missing all the weird stuff the actual compilers set.

Reply:
> I think these compiler overrides are necessary unless the `nci-openmpi` package exports its paths differently

Do you think that is something the `nci-openmpi` package should do? Otherwise this fix might need to be done for many packages. 

Is there a standard** for this sort of thing?


** generally assumed way of doing things

Reply:
It's pretty much detailed here. Following that, I should probably set `CC` and go in the environment, rather than just as arguments to configure. https://spack.readthedocs.io/en/latest/packaging_guide.html#mpi-support-in-spack

Reply:
Alright, I did try that recommendation, and no, it doesn't work. I think passing the compilers through to configure is the way to do it. I don't know that there's a "standard", but there's a mix between packages defining the compilers in the configure stage, or in the environment at the install stage. So I think both are legitimate approaches.

Reply:
Let's go with what works, for now.